tests/core/DependencyTrackerTest.py

```diff
@@ -110,6 +110,32 @@ def test_get_all_dependencies(self):
         self.assertEqual(uut.get_all_dependencies(3), {0, 1})
         self.assertEqual(uut.get_all_dependencies(4), {0, 2})
 
+    def test_dependants(self):
+        uut = DependencyTracker()
+
+        self.assertEqual(uut.dependants, set())
+
+        uut.add(0, 1)
+        uut.add(0, 2)
+        uut.add(1, 3)
+        uut.add(1, 3)
+        uut.add(4, 5)
+
+        self.assertEqual(uut.dependants, {1, 2, 3, 5})
+
+    def test_dependencies(self):
+        uut = DependencyTracker()
+
+        self.assertEqual(uut.dependencies, set())
+
+        uut.add(1, 2)
+        uut.add(2, 3)
+        uut.add(3, 4)
+        uut.add(3, 5)
+        uut.add(2, 5)
+
+        self.assertEqual(uut.dependencies, {1, 2, 3})
```

Comment:
I think the doctests are enough. Do you want to have some specific corner case tested?